src/import.js

```diff
@@ -250,6 +250,69 @@ padlock.import = (function(crypto, DisposableSource, Papa) {
         return data.split("\n")[0] == "url,username,password,extra,name,grouping,fav";
     }
 
+    /**
+     * Checks if a given string represents a KeePassX exported XML file
+     */
+    function isKeePassXML(xmldata) {
+        return xmldata.split("\n")[0] == "<!DOCTYPE KEEPASSX_DATABASE>";
+    }
+
+    /**
+     * Imports data from the XML file exported by KeePassX
+     */
+     function importKeePassXML(xmldata) {
+
+         // parse the xmldata string into actual XML format
+         parser = new DOMParser();
+         xmlDoc = parser.parseFromString(xmldata,"text/xml");
+
+         // initialize list of all records in this XML file
+         var records = [];
+
+         // get all the "groups" (aka categories) from xmlDoc and loop through them
+         var groups = xmlDoc.getElementsByTagName("group");
+         for (group = 0; group < groups.length; group++) {
```

Comment:
The `group` variable should be initialised as a local variable, i.e. `var group = 0`. Also, maybe use a variable name that indicates its an index, not the actual xml node. How about just using `i`?